File: checks/python/src/section508/checks/error_suggestion.py
import re

from section508.dom import SPACE, text_of

WEAK = re.compile(r"^(invalid\.?|invalid date\.?|error\.?|required\.?|not valid\.?)$", re.I)
HINT = re.compile(r"like |use |example|yyyy|year-month|@ |format|must include|try |include an? ", re.I)


def check(root):
    if not root:
        return []

    text = SPACE.sub(" ", text_of(root)).strip()
    if not text:
        return ["There is no suggestion in text."]
    if WEAK.match(text) or not HINT.search(text):
        return ["The message does not suggest how to fix the error."]
    return []

File: checks/ruby/lib/section508/checks/error_suggestion.rb
# frozen_string_literal: true

module Section508
  module Checks
    # When the format is known, the text says how to fix it.
    class ErrorSuggestion
      WEAK = /^(invalid\.?|invalid date\.?|error\.?|required\.?|not valid\.?)$/i
      HINT = /like |use |example|yyyy|year-month|@ |format|must include|try |include an? /i

      def self.call(root)
        return [] unless root

        text = Dom.norm(root)
        return ["There is no suggestion in text."] if text.empty?
        return ["The message does not suggest how to fix the error."] if WEAK.match?(text) || !HINT.match?(text)

        []
      end
    end
  end
end
